improver_tests/precipitation/snow_probability_at_surface/test_snow_probability_at_surface.py

```diff
@@ -0,0 +1,50 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Unit tests for SnowProbabilityAtSurface plugin"""
+
+import numpy as np
+import pytest
+
+from improver.precipitation.snow_probability_at_surface import (
+    SnowProbabilityAtSurface,
+)
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+
+ATTRIBUTES = {
+    "institution": "Met Office",
+    "source": "Met Office Unified Model",
+    "title": "UKV Forecast on UK 2 km Standard Grid",
+}
+
+
+@pytest.fixture()
+def wet_bulb_temperature_integral():
+    """Create a test cube of wet bulb temperature integral at surface"""
+    data = np.full(
+        (2, 3), 100, dtype=np.float32
+    )  # dummy data that will be overriden in tests
+    wet_bulb_temp_at_surf = set_up_variable_cube(
+        data,
+        name="wet_bulb_temperature_integral",
+        units=1,
+        attributes=ATTRIBUTES,
+    )
+
+    return wet_bulb_temp_at_surf
+
+
+@pytest.mark.parametrize(
+    "wb_temp_data , expected_probability",
+    ((-3, 1), (0, 1), (56.25, 0.75), (112.5, 0.5), (168.75, 0.25), (225, 0), (255, 0)),
```

Comment:
Please can you add a test case showing what happens when the data are masked? I believe we should expect masked data to be returned.